gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+        } else if (CsNullFlavor.UNK.value().equals(center)) {
+            return null;
+        } else if (validValue(availabilityTimeValue)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(availabilityTimeValue));
+        }
+
+        return null;
+    }
+
+    private String getTSStringValue(TS ts) {
+        if (ts == null) {
+            return null;
+        } else if (ts.getValue() != null) {
+            return ts.getValue();
+        } else if (ts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private String getIVXBTSStringValue(IVXBTS ivxbts) {
+        if (ivxbts == null) {
+            return null;
+        } else if (ivxbts.getValue() != null) {
+            return ivxbts.getValue();
+        } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private boolean validValue(String value) {
+        return value != null && !CsNullFlavor.UNK.value().equals(value);
+    }
+
+    private EncounterParticipantComponent getRecorder(RCMRMT030101UK04Author author) {
+        var recorder = new EncounterParticipantComponent();
+        var coding = new Coding();
+
+        coding
+            .setSystem(RECORDER_SYSTEM)
+            .setCode(RECORDER_CODE)
+            .setDisplay(RECORDER_DISPLAY);
+
+        return recorder
+            .addType(new CodeableConcept(coding))
+            .setIndividual(new Reference(PRACTITIONER_REFERENCE_PREFIX + author.getAgentRef().getId().getRoot()));
+    }
+
+    private boolean isNonNullParticipant2(RCMRMT030101UK04Participant2 participant2) {
+        return participant2.getNullFlavor() == null;
+    }
+
+    private EncounterParticipantComponent getPerformer(RCMRMT030101UK04Participant2 participant2) {
+        var performer = new EncounterParticipantComponent();
+        var coding = new Coding();
+
+        coding
+            .setSystem(PERFORMER_SYSTEM)
+            .setCode(PERFORMER_CODE)
+            .setDisplay(PERFORMER_DISPLAY);
+
+        return performer
+            .addType(new CodeableConcept(coding))
+            .setIndividual(new Reference(PRACTITIONER_REFERENCE_PREFIX + participant2.getAgentRef().getId().getRoot()));
+    }
+
+    private List<EncounterParticipantComponent> getEncounterParticipant(RCMRMT030101UK04Author author,
+        List<RCMRMT030101UK04Participant2> participant2List) {
+        List<EncounterParticipantComponent> participantList = new ArrayList<>();
+
+        if (author.getNullFlavor() == null) {
+            participantList.add(getRecorder(author));
+        }
+
+        // TODO: If author has a nullFlavor then create a recorder which references the Unknown Practitioner (NIAD-2026)
+
+        var participant2 = participant2List.stream().filter(this::isNonNullParticipant2).findFirst();
```

Comment:
minor - an `ifPresent` could be chained here and we could do `.ifPresent(participant ->[add to whatever]);` to get rid of the separate if statement